repos/spack_repo/builtin/packages/openmx/package.py

```diff
@@ -86,11 +112,17 @@ def common_arguments(self):
             fc_option.extend([self.compiler.openmp_flag, "-Ccpp"])
         else:
             common_option.append("-O3")
+            cc_option.extend(
+                ["-Wno-error=implicit-function-declaration", "-Wno-error=implicit-int", "-fcommon"]
+            )
```

Comment:
```suggestion
                ["-Wno-error=implicit-function-declaration", "-fcommon"]
            )
            for cc in ("aocc", "llvm"):
               if f"%{cc}" in spec:
                   if spec[cc].version >= Version("15.0"):
                       cc_option.append("-Wno-error=implicit-int")
```
Completely untested, and just an idea.

If compiling with older versions is not supported, then `conflicts()` for older `%aocc` and `%gcc` versions could be added instead.

And if/as `%llvm` is not supported, a conflicts `conflicts("%llvm")` would reject vanilla llvm as a compiler.